### DISCUSSION TITLE
dfu and nrf52840 dongle

### DISCUSSION THREAD

**Discussion by Yagoor** [hathach/tinyusb#937] [Q&A] [answered]
Hi,
I've been trying to trigger the dfu runtime, tud_dfu_runtime_reboot_to_dfu_cb, using the nrfutil but I've not been successful so far. 
Perhaps I'm missing something but using the dfu_runtime example, the device is not listed as a CDC device, so the nrfutil cannot find it.
So I took the cdc_msc example and extended it with the dfu_runtime example.
Now I have the device in listed however when trying to trigger the dfu reboot using the nrfutil I get the following error.

"No trigger interface found for device with serial number"

Has anyone tried to use the dfu_runtime with nrfutil? 
Thank you.

**Comment by hathach** [comment, marked as answer]
Are you sure nrfutil use dfu class ?

**Reply by Yagoor** [reply]
Thank you for the reply. I did not consider this case.

According to the nordic [documentations](https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_nrf5_v16.0.0%2Flib_dfu_trigger_usb.html), no.

In summary, this might be the issue:
"Since the DFU mode entered is not USB DFU 1.1 compliant, the interface class has been changed to "Vendor Specific" (0xFF). The subclass and protocol are both 0x01, as in USB DFU 1.1."

Am I able to change the tinyusb behavior to implement vendor specific functionality?
Using some weak function or some extensions?

**Reply by hathach** [reply]
you can use either use vendor class (currently only work if there is a pair of endpiont) or simply implement your own driver to implement though, there is note in readme mentioning how to do so.

**Reply by Yagoor** [reply]
Thank you for the reply.
I think I have enough to go on. :)

**Reply by hathach** [reply]
if you think your question is answered, don't forget to mark one of the reply (even your own as anwser), this will help other user with similar issue.
